x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/engines/delete_engine_modal.tsx

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React from 'react';
+
+import { useActions, useValues } from 'kea';
+
+import { EuiConfirmModal } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+
+import { EnginesListLogic } from './engines_list_logic';
+
+export const DeleteEngineModal: React.FC = () => {
+  const { closeDeleteEngineModal, deleteEngine } = useActions(EnginesListLogic);
+  const {
+    deleteModalEngineName: engineName,
+    isDeleteModalVisible,
+    isDeleteLoading,
+  } = useValues(EnginesListLogic);
+
+  if (isDeleteModalVisible) {
+    return (
+      <EuiConfirmModal
+        title={i18n.translate('xpack.enterpriseSearch.content.engineList.deleteEngineModal.title', {
+          defaultMessage: 'Permanently delete this engine?',
+        })}
+        onCancel={() => {
+          closeDeleteEngineModal();
+        }}
+        onConfirm={() => {
+          deleteEngine({ engineName });
+        }}
+        cancelButtonText="No, cancel"
+        confirmButtonText={i18n.translate(
+          'xpack.enterpriseSearch.content.engineList.deleteEngineModal.confirmButton.title',
+          {
+            defaultMessage: 'Yes, Delete this engine',
+          }
+        )}
+        buttonColor="danger"
+        defaultFocusedButton="confirm"
```

Comment:
question: Do we want to make the confirm button the default focus? Would make it very easy for folks to accidentally click through a destructive action.

Reply:
I think I add this based on existing delete indices modal. 
@sphilipse would you suggest changing default focus to `cancel` or remove default focus all together?

Reply:
I think remove the defaultFocusedButton property, and it'll autofocus on the close modal button.